lib/AST/ASTImporterLookupTable.cpp

```diff
@@ -27,33 +27,28 @@ struct Builder : RecursiveASTVisitor<Builder> {
     LT.add(D);
     return true;
   }
-  // Friend declarations introduce new types, which should be visible during
-  // lookup of a specific named declaration even if it would be illegal to
-  // reference them at the root level of the TU.
-  // This visitor is necessary because in case of some friend declarations the
-  // AST does not contain the referenced friend class or function as a child of
-  // the referencing declaration, thus causing the recursive discovery fail.
-  // This visitor helps this discovery process by making the lookup table aware
-  // of these "hidden" declarations.
+  // In most cases the FriendDecl inside the referencing class contains the
+  // declaration of the "befriended class" as a child node, so it is discovered
+  // during the recursive visitation. Dependent types behave this way. In some
+  // other cases the "befriended class" must be visited explicitly, and this is
+  // simulated by adding its declaration to the lookup table.
```

Comment:
Some more nitpicking, but we are very close now: So, this is not a "simulation". I'd rather say something like this `... the non-child "befriended class" must be fetched explicitly from the FriendDecl and only then can we add that to the lookup table`.